### PR TITLE
Settings for weekplanner feature 857

### PR DESCRIPTION
**Pull request by Taoshix**
# Description

Added Number of Actitivies to Display, Show Only Actitivies and Show Settings for Citizens to the api client to be used in the weekplanner.

Fixes related to #857 in weekplanner

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Using the new features do show up correctly in the database with no errors. There are unit tests in the weekplanner and api client that passes with data received from the database.

**Development Configuration**

* Flutter version: 2.0.5
* Dart version: 2.12.3

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: lib/http/http_client.dart [hunk 1]
```diff
@@ -110,6 +110,7 @@ class HttpClient implements Http {
         json = jsonDecode(res.body);
 
         if (res.statusCode > 300) {
+          print(res.statusCode);
```
Comment (Jaffenheimer) [inline lib/http/http_client.dart]: Meningen dette skal være der eller brugt til debug?

File: lib/models/settings_model.dart [hunk 2]
```diff
@@ -86,7 +92,10 @@ class SettingsModel implements Model {
     greyscale = settingsJson['greyScale'] == 1;
     lockTimerControl = settingsJson['lockTimerControl'] == 1;
     pictogramText = settingsJson['pictogramText'] == 1;
-    showPopup = settingsJson['showPopup'] == 0;
+    showPopup = settingsJson['showPopup'] == 1;
```
Comment (Jaffenheimer) [inline lib/models/settings_model.dart:82]: Hvorfor bliver denne ændret fra true til false? Vil den ikke vise popup hvis den er false eller?

File: lib/offline_database/offline_db_handler.dart [hunk 3]
```diff
@@ -515,14 +528,15 @@ class OfflineDbHandler {
         <String>[userId])).first['settingsId'];
 
     await db.rawUpdate('''UPDATE Settings SET
-        id = ?,
```
Comment (Jaffenheimer) [inline lib/offline_database/offline_db_handler.dart:519]: Hvorfor har i fjernet ID'et her? Skal den ikke bruges mere eller, har i tilføjet den andetsteds jeg ikke lige kan finde?

**Review by Jaffenheimer — COMMENTED** [review]
Har lige et par spørgsmål til denne :)

File: lib/offline_database/offline_db_handler.dart (hunk 3 above)
Comment (Taoshix) [reply lib/offline_database/offline_db_handler.dart:519]: Den skal ikke bruges når man skal opdatere settings, da det er en primary key som ikke skal ændres

File: lib/http/http_client.dart (hunk 1 above)
Comment (Taoshix) [reply lib/http/http_client.dart]: brugt til debug, skal nok fjerne det i næste commit

File: lib/models/settings_model.dart (hunk 2 above)
Comment (Taoshix) [reply lib/models/settings_model.dart:82]: Den bliver sat fra false til true, så det er meningen at den skal vise pop ups som standard

**Review by Jaffenheimer — DISMISSED** [review]
Looks good to me now after the answers to the questions and the removal of debug print

**Review by Jaffenheimer — APPROVED** [review]
Approved again since last was dismissed because of fix

**Review by AOsnes — APPROVED** [review]
looks good to me

**Pull request merged**